File: 967-numbers-with-same-consecutive-differences/967-numbers-with-same-consecutive-differences.cpp
class Solution {
public:
    set<int> st;
    
    void recur(int n, int k, int t){
        if(n==0){
            st.insert(t);
            return;
        }
        int b = t%10;
        t*=10;
        int nxt = k+b;
        if(k+b<10){
            t+=nxt;
            recur(n-1,k,t);
            t/=10;
            t*=10;
        }
        nxt = b-k;
        if(nxt>=0){
            t+=nxt;
            recur(n-1,k,t);
        }
    }
    
    vector<int> numsSameConsecDiff(int n, int k) {
        vector<int> res;
        for(int i=1;i<=9;i++){
            recur(n-1,k,i);
        }
        for(auto it:st)res.push_back(it);
        return res;
    }
};

File: 967-numbers-with-same-consecutive-differences/967-numbers-with-same-consecutive-differences.py
class Solution:
    def numsSameConsecDiff(self, n: int, k: int) -> List[int]:
        res = [1,2,3,4,5,6,7,8,9]
        
        for i in range(0,n-1):
            res2=  []
            for x in res:
                y = x%10;
                if y+k<10 :
                    res2.append(x*10+y+k)
                if k>0 and y-k>=0:
                    res2.append(x*10+y-k)
            res=res2
            
        return res